theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -179,7 +179,10 @@ def _optimize_loop(
             # do optimizer step
             self.linear_solver.linearization.linearize()
             try:
-                delta = self.compute_delta(**kwargs)
+                if truncated_grad_loop:
```

Comment:
how about
```
if backward_mode == BackwardMode.IMPLICIT or backward_mode == BackwardMode.TRUNCATED
```
so we have less redundant variables?

Reply:
Agreed.

Reply:
Made #91 so we make this in a new PR, because it will create conflicts with several currently opened PRs.